src/components/FullscreenLoadingIndicator.js

```diff
@@ -25,18 +30,42 @@ const defaultProps = {
  * @param {Object} props
  * @returns {JSX.Element}
  */
-const FullScreenLoadingIndicator = (props) => {
-    if (!props.visible) {
-        return null;
+class FullScreenLoadingIndicator extends React.Component {
+    componentDidMount() {
+        if (!this.props.name || !this.props.timeout) {
```

Comment:
**suggestion:** We could still log without additional context. But if we want to enforce additional context then maybe we should throw and return so that someone implementing this will remember to add details.

Reply:
Good point in order to distinct the loader, we should require some context - at least the screen name where it's used, so we should throw an error - if someone tries to setup a new or existing loader with logging incorrectly they'll know right away

Reply:
Updated the logic to throw an error if `props.logDetail` exists but `logDetail.name` is not set

But since `logDetail.name` is flagged as `PropTypes.required` I don't know whether we need to throw an exception, and maybe we should remove it and just return